graph.cpp

```diff
@@ -0,0 +1,103 @@
+#include <iostream>
+#include <list>
+#include <map>
+#include <set>
+
+
+
+class Node{
+    public:
+        int nodeID;
+        // Placeholder for 3D location info as I'm not sure how Godot handles it
+        std::tuple <double, double, double> location;
+        // Placeholder for formula data
+        int data;
+        // Potentially split into seperate parent/child sets?
+        std::set<Node*> neighbors;
+
+        Node(){
+            nodeID = 0;
+            location = std::make_tuple(0,0,0);
+            data = 0; 
+        }
+        Node(int nodeID, const std::tuple<double, double, double> location, int data){
```

Comment:
Const ref not just const
```suggestion
        Node(int nodeID, const std::tuple<double, double, double>& location, int data){
```

Reply:
Apologies, haven't worked in C in a while and its my first program in C++, just a bit rusty

Reply:
All good, for a first program in C++ this is very impressive. Passing by const reference is like passing by const pointer but handwaving away the address and dereference logic. [Here's a quick two minute read on passing by value vs passing by reference in C++](https://www.quantstart.com/articles/Passing-By-Reference-To-Const-in-C/)